inc/zoo/swar/SWAR.h

```diff
@@ -85,15 +85,24 @@ struct SWAR {
         LowerBits = MostSignificantBit - LeastSignificantBit,
         MaxUnsignedLaneValue = LeastSignificantLaneMask;
 
-    template <typename U>
-    constexpr static auto from_array(const U (&values)[Lanes]) noexcept {
+    template <typename InputIt>
+    constexpr static auto from_range(InputIt first, InputIt last) noexcept {
         auto result = T{0};
-        for (auto value : values) {
-            result = (result << NBits) | value;
+        for (; first != last; ++first) {
+            result = (result << NBits) | *first;
         }
-        return SWAR{result};
+        return result;
     }
 
+    template <typename U>
+    constexpr static auto from_array(const U (&values)[Lanes]) noexcept {
+        return SWAR{from_range(std::begin(values), std::end(values))};
```

Comment:
NOOOOOO!!!!
do not qualify `begin` and `end` with `std::`!!!!!
We define `begin` and `end` within zoo for zoo's own containers.
The same with `swap`.
They are library customization points.
The correct way is to do this, whenever possible, it is fine if you don't do this whenever not possible:
```c++
constexpr static auto from_array(const U (&values)[Lanes]) noexcept {
    using std::begin; using std::end; // fallback to std
    yada yada begin(range), yada yada end(range)
```